.github/workflows/build.yaml

```diff
@@ -0,0 +1,94 @@
+name: Build
+
+on:
+  pull_request:
+    branches: [ pr/optimize-workflows ]
```

Comment:
need to update the branch to main and release-* once the review is complete (using this branch to test the workflows)

Reply:
updated, ready for review/merge